etc/scripts/timetable/old-stack-import/django/django-export-events-command.py

```diff
@@ -0,0 +1,374 @@
+        module = self.get_module(event)
+        return module.fullname
+
+
+class ModuleShortNameColumnSpec(ColumnSpec):
+    name = "Module Short Name"
+
+    def extract_value(self, event):
+        module = self.get_module(event)
+        return module.name
+
+
+class SeriesIdColumnSpec(ColumnSpec):
+    name = "Series Name"
+
+    def extract_value(self, event):
+        series = self.get_series(event)
+        return series.id
+
+class SeriesNameColumnSpec(ColumnSpec):
+    name = "Series Name"
+
+    def extract_value(self, event):
+        series = self.get_series(event)
+        return series.title
+
+
+class EventAttrColumnSpec(ColumnSpec):
+    attr_name = None
+
+    def get_attr_name(self):
+        assert self.attr_name is not None
+        return self.attr_name
+
+    def extract_value(self, event):
+        return getattr(event, self.get_attr_name())
+
+
+class EventIdColumnSpec(EventAttrColumnSpec):
+    name = "Event ID"
+    attr_name = "id"
+
+class EventTitleColumnSpec(EventAttrColumnSpec):
+    name = "Title"
+    attr_name = "title"
+
+
+class EventLocationColumnSpec(EventAttrColumnSpec):
+    name = "Location"
+    attr_name = "location"
+
+
+class EventUidColumnSpec(ColumnSpec):
+    name = "UID"
+
+    def extract_value(self, event):
+        return event.get_ical_uid()
+
+
+class EventDateTimeColumnSpec(ColumnSpec):
+    timezone = pytz.timezone("Europe/London")
+
+    def get_datetime_utc(self, event):
+        raise NotImplementedError()
+
+    def extract_value(self, event):
+        dt_utc = self.get_datetime_utc(event)
+        return self.timezone.normalize(dt_utc.astimezone(self.timezone)).isoformat()
+
+
+class EventStartDateTimeColumnSpec(EventDateTimeColumnSpec):
+    name = "Start"
+
+    def get_datetime_utc(self, event):
+        return event.start
+
+
+class EventEndDateTimeColumnSpec(EventDateTimeColumnSpec):
+    name = "End"
+
+    def get_datetime_utc(self, event):
+        return event.end
+
+
+class EventMetadataColumnSpec(ColumnSpec):
+    metadata_path = None
+
+    def get_metadata_path(self):
+        if self.metadata_path is None:
+            raise ValueError("no metadata_path value provided")
+        return self.metadata_path
+
+    def extract_value(self, event):
+        metadata = event.metadata
+
+        segments = self.get_metadata_path().split(".")
+        for i, segment in enumerate(segments):
+            if not isinstance(metadata, dict):
+                return None
+            if i == len(segments) - 1:
+                return metadata.get(segment)
+            metadata = metadata.get(segment)
+
+
+class EventTypeColumnSpec(EventMetadataColumnSpec):
+    name = "Type"
+    metadata_path = "type"
+
+
+class EventLecturerColumnSpec(EventMetadataColumnSpec):
+    name = "People"
+    metadata_path = "people"
+
+    def extract_value(self, event):
+        value = super(EventLecturerColumnSpec, self).extract_value(event)
+        return None if value is None else ", ".join(value)
```

Comment:
Unfortunately there aren't enough docs in here to reasonably review what's going on. It's generating CSV, and since I don't have any means to test that it works, I'm not sure it's super important for me to review aside from just understand the process.